apps/prairielearn/src/middlewares/authzCourseOrInstance.sql

```diff
@@ -1,6 +1,20 @@
 -- BLOCK select_authz_data
+WITH
+  access_mode AS (
+    SELECT
+      mode,
+      mode_reason
+    FROM
+      ip_to_mode ($ip, $req_date, $user_id)
+  )
```

Comment:
I could also just join on `ip_to_mode`?

Reply:
I don't think it really matters either way. We normally JOIN, but this is also fine if you think it's clearer.

Reply:
Switched to JOINing in a8f961a.